Source/UI/Discover/DiscoverView.swift

```diff
@@ -0,0 +1,167 @@
+        .toolbar {
+            ToolbarItemGroup(placement: .navigationBarTrailing) {
+                Button {
+                    helpDrawerState.isShowingDiscoverHelpDrawer = true
+                } label: {
+                    Image.navIconHelp
+                }
+                .popover(isPresented: $helpDrawerState.isShowingDiscoverHelpDrawer) {
+                    Group {
+                        if #available(iOS 16.0, *) {
+                            HelpDrawerCoordinator.helpDrawerView(for: self) {
+                                helpDrawerState.isShowingDiscoverHelpDrawer = false
+                            }
+                            .presentationDetents([.medium])
+                        } else {
+                            HelpDrawerCoordinator.helpDrawerView(for: self) {
+                                helpDrawerState.isShowingDiscoverHelpDrawer = false
+                            }
+                        }
+                    }
+                    .onAppear {
+                        Analytics.shared.trackDidShowScreen(screenName: helpDrawerType.screenName)
+                        HelpDrawerCoordinator.didShowHelp(for: helpDrawerType)
+                    }
+                }
+                Button {
+                    Analytics.shared.trackDidTapButton(buttonName: "compose")
+                    showCompose()
+                } label: {
+                    Image.navIconWrite
+                }
+            }
+        }
+        .onReceive(NotificationCenter.default.publisher(for: .didChangeDiscoverFeedAlgorithm)) { _ in
+            Task {
+                await dataSource.loadFromScratch()
+            }
+        }
+        .onReceive(NotificationCenter.default.publisher(for: .didUpdateRelationship)) { _ in
+            Task {
+                await dataSource.loadFromScratch()
+            }
+        }
+        .onReceive(NotificationCenter.default.publisher(for: .didPublishPost)) { _ in
+            Task {
+                await dataSource.loadFromScratch()
+            }
+        }
+        .onAppear {
+            CrashReporting.shared.record("Did Show Discover")
+            Analytics.shared.trackDidShowScreen(screenName: "discover")
+            HelpDrawerCoordinator.showFirstTimeHelp(for: helpDrawerType, state: helpDrawerState)
+        }
+    }
+
+    private func showCompose() {
+        let controller = NewPostViewController()
+        controller.didPublish = { post in
+            NotificationCenter.default.post(.didPublishPost(post))
+        }
+        let navController = UINavigationController(rootViewController: controller)
+        appController.present(navController, animated: true)
+    }
+}
+
+class SearchTextFieldObserver: ObservableObject {
+    @Published
+    var debouncedText = ""
+
+    @Published
+    var text = ""
+
+    private var subscriptions = Set<AnyCancellable>()
+
+    init() {
+        $text
+            .removeDuplicates()
```

Comment:
I tested it and it doesn't trigger a new search, and moving removeDuplicates() to come after debounce() doesn't make a difference. I cannot explain why because combine is a bit automagical, but removeDuplicates() was called before in UniversalSearchResults (and we didn't find any issue with that) and it is called before in most examples I found in the internet. So I'm leaving as it is as it has more testing behind its shoulders.